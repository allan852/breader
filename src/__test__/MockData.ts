import { IntlProvider } from 'react-intl'
import { messages } from '../locales'
import { IArticle, IFeed } from '../schemas'

export const intlProviderProps: IntlProvider.Props = {
  locale: 'en-US',
  messages: messages['en-US'],
}

const intlProvider = new IntlProvider(intlProviderProps, {})
export const intl = intlProvider.getChildContext().intl

export const feed: IFeed = {
  _id: 'http://www.ruanyifeng.com/blog/atom.xml',
  _rev: '5-55772bf2b4d4b74608a60ac3480334e4',
  author: 'ruanyifeng',
  categories: [],
  createTime: 1563686468437,
  deleteTime: 0,
  description: "Ruan YiFeng's Blog",
  etag: '"11d86-58e2cc1eab031"',
  favicon: 'http://www.ruanyifeng.com/favicon.ico',
  generator: 'Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)',
  language: '',
  link: 'http://www.ruanyifeng.com/blog/',
  publishTime: 1563697765000,
  time: 1563697765000,
  title: '阮一峰的网络日志',
  url: 'http://www.ruanyifeng.com/blog/atom.xml',
}

export const article: IArticle = {
  _id: 'tag:www.ruanyifeng.com,2019:/blog//1.2100',
  _rev: '5-ff6fc450616071a8f7d022b7532e85aa',
  author: 'ruanyifeng',
  categories: [],
  comments: 'article comments',
  createTime: 1563697765000,
  deleteTime: 0,
  description: 'article description',
  enclosures: [],
  feedId: 'http://www.ruanyifeng.com/blog/atom.xml',
  image: '',
  isDayFirst: true,
  isStarred: false,
  isUnread: false,
  link: 'http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html',
  originLink: '',
  publishTime: 1563697765000,
  summary: 'article summary',
  time: 1563697765000,
  title: 'article title',
}
