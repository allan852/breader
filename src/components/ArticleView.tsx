import { Empty, Icon } from 'antd'
import Immutable from 'immutable'
import React, { Component, PureComponent } from 'react'
import { InjectedIntlProps, injectIntl } from 'react-intl'
import greyLogo from '../images/grey-logo.png'
import { IArticle, IFeed } from '../schemas'
import Utils from '../utils'
import WebviewDrawer from './WebviewDrawer'

import '../styles/ArticleView.less'

export interface IArticleViewOwnProps {
}

export interface IArticleViewReduxState {
    currentArticle: IArticle | null
    articles: Immutable.List<IArticle>
    feedsMap: Immutable.Map<string, IFeed>
}

export interface IArticleViewReduxDispatch {
    asyncStarArticle: (articleId: string, isStarred: boolean) => Promise<void>
}

interface IArticleViewProps extends IArticleViewOwnProps, IArticleViewReduxDispatch, IArticleViewReduxState {
}

interface IArticleViewState {
    hoverLink: string
    isWebviewDrawerVisible: boolean
    webviewDrawerSrc: string,
    isStarredsMap: {[_id: string]: boolean},
}

class ArticleView extends PureComponent<IArticleViewProps & InjectedIntlProps> {
    public state: IArticleViewState
    private _articleContentIsAppended: boolean
    private _articleContentElement: HTMLDivElement
    private _articleContentLinks: string[]
    public constructor(props: IArticleViewProps & InjectedIntlProps) {
        super(props)
        this.state = {
            hoverLink: '',
            isStarredsMap: {},
            isWebviewDrawerVisible: false,
            webviewDrawerSrc: '',
        }
        this._articleContentIsAppended = false
        this._articleContentElement = document.createElement('div')
        this._articleContentLinks = []
    }
    public componentWillReceiveProps(props: IArticleViewProps) {
        const currentArticle = props.currentArticle
        const isStarredsMap = this.state.isStarredsMap
        if (currentArticle && currentArticle !== this.props.currentArticle) {
            this._parseArticleContent(currentArticle.description)
            if (isStarredsMap[currentArticle._id] === undefined) {
                isStarredsMap[currentArticle._id] = currentArticle.isStarred
                this.setState({
                    isStarredsMap: {...isStarredsMap},
                })
            }
        }
    }
    public componentDidUpdate() {
        if (this._articleContentIsAppended) {
            return
        }
        this._articleContentIsAppended = true
        const div = document.querySelector('.article-content')
        if (div && this._articleContentElement) {
            while (div.firstChild) {
                div.removeChild(div.firstChild);
            }
            div.appendChild(this._articleContentElement)
            const view = div.closest('.view-content')
            if (view) {
                view.scrollTo(0, 0)
            }
        }
    }
    public handleStarIconClick = () => {
        const currentArticle = this.props.currentArticle
        const isStarredsMap = this.state.isStarredsMap
        if (currentArticle) {
            const articleId = currentArticle._id
            isStarredsMap[articleId] = !isStarredsMap[articleId]
            this.setState({
                isStarredsMap: { ...isStarredsMap },
            })
            this.props.asyncStarArticle(articleId, isStarredsMap[articleId])
        }
    }
    public handleContentClick = (e: any) => {
        const link = this.state.hoverLink
        if (link) {
            this.setState({
                isWebviewDrawerVisible: true,
                webviewDrawerSrc: link,
            })
        }
    }
    public handelWebviewClose = (e: any) => {
        this.setState({ isWebviewDrawerVisible: false })
    }
    public handleMouseLeave = () => {
        this.setState({hoverLink: ''})
    }
    public handleMouseOverInfo = (link: string) => {
        if (link) {
            this.setState({hoverLink: link})
        }
    }
    public handleMouseOverContent = (e: any) => {
        const target = e.target
        if (target.tagName === 'A' && target.dataset.index) {
            const link = this._articleContentLinks[target.dataset.index]
            this.setState({hoverLink: link})
        } else {
            this.setState({hoverLink: ''})
        }
    }
    public render() {
        let viewContent: any
        let starIcon: any

        if (this.props.currentArticle) {
            const article = this.props.currentArticle
            const feedsMap = this.props.feedsMap
            const intl = this.props.intl
            const feed = feedsMap.get(article.feedId)
            viewContent = (
                <div className="view-content" onMouseLeave={this.handleMouseLeave} onClick={this.handleContentClick}>
                    <div className="article-info" onMouseOver={() => this.handleMouseOverInfo(article.link)}>
                        <div className="article-date"><p>{Utils.timeToDateTimeString(article.time)}</p></div>
                        <div className="article-title"><h1>{article.title}</h1></div>
                        <div className="article-author">
                            <p>{article.author} @ {feed ? feed.title : intl.formatMessage({id: 'unknown'})}</p>
                        </div>
                    </div>
                    <div className="article-content" onMouseOver={this.handleMouseOverContent}>{' '}</div>
                </div>
            )
            if (this.state.isStarredsMap[(article._id as string)]) {
                starIcon = (<Icon type="star" theme="filled"
                    onClick={this.handleStarIconClick} />)
            } else {
                starIcon = (<Icon type="star" theme="outlined"
                    onClick={this.handleStarIconClick} />)
            }
        } else {
            viewContent = (
                <div className="view-content">
                    <div style={{'marginTop': '128px'}}>
                        <Empty image={greyLogo} description={' '} />
                    </div>
                </div>
            )
            starIcon = null
        }
        return (
            <div className="article-view" >
                <div className="view-header">
                    <div className="view-header-right">
                        {starIcon}
                    </div>
                </div>
                {viewContent}
                <div className="view-footer"><p>{this.state.hoverLink ? 'Open ' + this.state.hoverLink : ''}</p></div>
                <WebviewDrawer width={'calc(100vw - 490px)'} onClose={this.handelWebviewClose} visible={this.state.isWebviewDrawerVisible} src={this.state.webviewDrawerSrc} />
            </div>
        )
    }
    private _parseArticleContent(content: string) {
        const div = document.createElement('div')
        div.innerHTML = content
        const scripts = div.querySelectorAll('scripts')
        scripts.forEach((script) => {
            script.remove()
        })

        const links = div.querySelectorAll('a')
        this._articleContentLinks = []
        links.forEach((link, i) => {
            this._articleContentLinks.push(link.href)
            link.dataset.index = i + ''
            link.href = 'javascript:void(0);'
        })

        const frames = div.querySelectorAll('iframe')
        frames.forEach((frame) => {
            frame.setAttribute('sandbox', '')
        })
        this._articleContentElement = div
        this._articleContentIsAppended = false
    }
}

export default injectIntl(ArticleView)
