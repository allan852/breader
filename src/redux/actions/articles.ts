import { IArticle, IReduxAction } from '../../schemas'

export const ArticlesActionTypes = {
    SET_ARTICLES: 'SET_ARTICLES',
    SET_ARTICLES_FILTER: 'SET_ARTICLES_FILTER',
    SET_CURRENT_ARTICLE: 'SET_CURRENT_ARTICLE',

    ASYNC_FETCH_ARTICLES: 'ASYNC_FETCH_ARTICLES',
    ASYNC_FILTER_ARTICLES: 'ASYNC_FILTER_ARTICLES',
    ASYNC_SELECT_AND_READ_ARTICLE: 'ASYNC_SELECT_AND_READ_ARTICLE',
    ASYNC_SET_ALL_ARTICLES_READ: 'ASYNC_SET_ALL_ARTICLES_READ',
    ASYNC_STAR_ARTICLE: 'ASYNC_STAR_ARTICLE',
}

export interface ISetCurrentArticlePayload { article: IArticle | null }
export const setCurrentArticleAction = (article: IArticle | null): IReduxAction<ISetCurrentArticlePayload> => ({
    payload: { article },
    type: ArticlesActionTypes.SET_CURRENT_ARTICLE,
})

export interface ISetArticlesFilterPayload { filter: string }
export const setArticlesFilterAction = (filter: string): IReduxAction<ISetArticlesFilterPayload> => ({
    payload: { filter },
    type: ArticlesActionTypes.SET_ARTICLES_FILTER,
})

export interface ISetArticlesPayload { articles: IArticle[]}
export const setArticlesAction = (articles: IArticle[]): IReduxAction<ISetArticlesPayload> => ({
    payload: { articles },
    type: ArticlesActionTypes.SET_ARTICLES,
})

export const asyncFetchArticlesAction = (): IReduxAction<null> => ({
    payload: null,
    type: ArticlesActionTypes.ASYNC_FETCH_ARTICLES,
})

export interface IAsyncFilterArticlesPayload { filter: string }
export const asyncFilterArticlesAction = (filter: string): IReduxAction<IAsyncFilterArticlesPayload> => ({
    payload: { filter },
    type: ArticlesActionTypes.ASYNC_FILTER_ARTICLES,
})

export interface IAsyncSelectAndReadArticlePayload { articleId: string, articleIndex: number }
export const asyncSelectAndReadArticleAction = (articleId: string, articleIndex: number): IReduxAction<IAsyncSelectAndReadArticlePayload> => ({
    payload: { articleId, articleIndex },
    type: ArticlesActionTypes.ASYNC_SELECT_AND_READ_ARTICLE,
})

export interface IAsyncStarArticlePayload { articleId: string, isStarred: boolean }
export const asyncStarArticleAction = (articleId: string, isStarred: boolean): IReduxAction<IAsyncStarArticlePayload> => ({
    payload: { articleId, isStarred },
    type: ArticlesActionTypes.ASYNC_STAR_ARTICLE,
})

export interface IAsyncSetAllArticlesReadPayload { articleIds: string[] }
export const asyncSetAllArticlesReadAction = (articleIds: string[]): IReduxAction<IAsyncSetAllArticlesReadPayload> => ({
    payload: { articleIds },
    type: ArticlesActionTypes.ASYNC_SET_ALL_ARTICLES_READ,
})
