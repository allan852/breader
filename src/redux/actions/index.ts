import { Dispatch } from 'redux'
import { IReduxAction, IReduxAsyncAction } from '../../schemas'

export function asyncActionDispatcher<Type = {}>(dispatch: Dispatch<IReduxAction>, action: IReduxAction) {
    return new Promise<Type>((resolve, reject) => {
        const asyncAction: IReduxAsyncAction = {
            ...action,
            reject,
            resolve,
        }
        dispatch(asyncAction)
    })
}

export * from './articles'
export * from './feeds'
export * from './menu'
