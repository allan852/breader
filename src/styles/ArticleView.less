@import './Values.less';

.article-view {
    -webkit-app-region: drag;
    position: relative;
    padding: 42px 0 42px 0;
    height: 100%;

    .view-header {
        .wrapper-header;
        .anticon {
            color: @black-text-color;
            &:hover {
                color: @black-text-color-light;
            }
        }
    }
    .view-header-right {
        .wrapper-hf-right;
        .anticon {
            margin-top: 13px;
            margin-right: 16px;
        }
    }
    .view-footer {
        .wrapper-footer;
        padding: 0 16px 0 16px;
        p {
            font-size: 14px;
            color: @black-text-color-dark;
            line-height: 41px;
            width: 100%;
            overflow: hidden;
            text-overflow: ellipsis;
            white-space: nowrap;
            margin: 0;
            // margin-bottom: 0;
        }
    }
    .view-content {
        -webkit-app-region: none;
        height: 100%;
        overflow-x: hidden;
        overflow-y: scroll;

        .article-info {
            cursor: pointer;
            border-radius: 4px;
            padding: 16px;
            max-width: 420px;
            
            margin: 32px auto 12px auto;
            &:hover {
                background-color: @list-selected-color;
            }
            .article-date p {
                color: @black-text-color-dark;
                font-size: 12px;
                margin-bottom: 4px;
            }
            .article-title h1 {
                font-size: 32px;
                line-height: 40px;
                margin-bottom: 6px;
            }
            .article-author p {
                color: @black-text-color-dark;
                font-size: 12px;
                margin-bottom: 2px;
            }
        }
        .article-content {
            padding: 16px;
            max-width: 420px;
            margin-right: auto;
            margin-left: auto;
            img, video, iframe {
                max-width: 100%;
            }
        }
    }
}
