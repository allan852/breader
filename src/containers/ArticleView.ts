import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { Dispatch } from 'redux'
import ArticleView, { IArticleViewOwnProps, IArticleViewReduxDispatch, IArticleViewReduxState } from '../components/ArticleView'
import { asyncActionDispatcher, asyncStarArticleAction } from '../redux/actions'
import { IReduxAction, IReduxState } from '../schemas'

const mapStateToProps: MapStateToProps<IArticleViewReduxState, IArticleViewOwnProps, IReduxState> = (state: IReduxState) => ({
    articles: state.articles.list,
    currentArticle: state.articles.current,
    feedsMap: state.feeds.map,
    isUpdatingCurrentArticle: state.articles.isUpdatingCurrent,
})

const mapDispatchToProps: MapDispatchToProps<IArticleViewReduxDispatch, IArticleViewOwnProps> = (dispatch: Dispatch<IReduxAction>) => ({
    asyncStarArticle: (articleId: string, isStarred: boolean) => asyncActionDispatcher(dispatch, asyncStarArticleAction(articleId, isStarred)),
})

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(ArticleView)
