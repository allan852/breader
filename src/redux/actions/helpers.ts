import { Dispatch } from 'redux'
import { IReduxAction, IReduxAsyncAction } from '../../schemas'

export function actionCreator<Payload>(type: string) {
  return (payload: Payload): IReduxAction<Payload> => ({
    payload,
    type,
  })
}

export function asyncActionDispatcher<Type = any>(
  dispatch: Dispatch<IReduxAction>,
  action: IReduxAction
) {
  return new Promise<Type>((resolve, reject) => {
    const asyncAction: IReduxAsyncAction = {
      ...action,
      reject,
      resolve,
    }
    dispatch(asyncAction)
  })
}
