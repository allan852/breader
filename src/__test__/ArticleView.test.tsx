import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider as ReduxProvider } from 'react-redux'
import TestRenderer from 'react-test-renderer'
import ArticleView from '../containers/ArticleView'
import store from '../redux'
import { setCurrentArticleAction } from '../redux/actions'
import { article, intlProviderProps } from './MockData'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('ArticleView Testing', () => {
    const dispatch = store.dispatch

    it('dom testing', () => {
        const wrapper = Enzyme.mount(
            <ReduxProvider store={store}>
                <IntlProvider {...intlProviderProps}>
                    <ArticleView />
                </IntlProvider>
            </ReduxProvider>
        )
        dispatch(setCurrentArticleAction(article))
        wrapper.update()
    })

    // it('snapshot testing', () => {
    //     const renderer = TestRenderer.create(
    //         <ReduxProvider store={store}>
    //             <IntlProvider {...intlProviderProps}>
    //                 <ArticleView />
    //             </IntlProvider>
    //         </ReduxProvider>
    //     )
    //     const tree = renderer.toJSON()
    //     expect(tree).toMatchSnapshot()
    // })
})
