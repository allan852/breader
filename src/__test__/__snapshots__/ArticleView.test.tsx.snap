// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleView Testing snapshot testing 1`] = `
<injectIntl(ArticleViewComponent)
  articles={
    Immutable.List [
      Object {
        "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
        "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
        "author": "ruanyifeng",
        "categories": Array [],
        "comments": "article comments",
        "createTime": 1563697765000,
        "deleteTime": 0,
        "description": "article description",
        "enclosures": Array [],
        "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
        "image": "",
        "isDayFirst": true,
        "isStarred": false,
        "isUnread": false,
        "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
        "originLink": "",
        "publishTime": 1563697765000,
        "summary": "article summary",
        "time": 1563697765000,
        "title": "article title",
      },
    ]
  }
  asyncStarArticle={[MockFunction]}
  currentArticle={
    Object {
      "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
      "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
      "author": "ruanyifeng",
      "categories": Array [],
      "comments": "article comments",
      "createTime": 1563697765000,
      "deleteTime": 0,
      "description": "article description",
      "enclosures": Array [],
      "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
      "image": "",
      "isDayFirst": true,
      "isStarred": false,
      "isUnread": false,
      "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
      "originLink": "",
      "publishTime": 1563697765000,
      "summary": "article summary",
      "time": 1563697765000,
      "title": "article title",
    }
  }
  feedsMap={
    Immutable.Map {
      "http://www.ruanyifeng.com/blog/atom.xml": Object {
        "_id": "http://www.ruanyifeng.com/blog/atom.xml",
        "_rev": "5-55772bf2b4d4b74608a60ac3480334e4",
        "author": "ruanyifeng",
        "categories": Array [],
        "createTime": 1563686468437,
        "deleteTime": 0,
        "description": "Ruan YiFeng's Blog",
        "etag": "\\"11d86-58e2cc1eab031\\"",
        "favicon": "http://www.ruanyifeng.com/favicon.ico",
        "generator": "Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)",
        "language": "",
        "link": "http://www.ruanyifeng.com/blog/",
        "publishTime": 1563697765000,
        "time": 1563697765000,
        "title": "阮一峰的网络日志",
        "url": "http://www.ruanyifeng.com/blog/atom.xml",
      },
    }
  }
  isUpdatingCurrentArticle={false}
>
  <ArticleViewComponent
    articles={
      Immutable.List [
        Object {
          "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
          "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
          "author": "ruanyifeng",
          "categories": Array [],
          "comments": "article comments",
          "createTime": 1563697765000,
          "deleteTime": 0,
          "description": "article description",
          "enclosures": Array [],
          "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
          "image": "",
          "isDayFirst": true,
          "isStarred": false,
          "isUnread": false,
          "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
          "originLink": "",
          "publishTime": 1563697765000,
          "summary": "article summary",
          "time": 1563697765000,
          "title": "article title",
        },
      ]
    }
    asyncStarArticle={[MockFunction]}
    currentArticle={
      Object {
        "_id": "tag:www.ruanyifeng.com,2019:/blog//1.2100",
        "_rev": "5-ff6fc450616071a8f7d022b7532e85aa",
        "author": "ruanyifeng",
        "categories": Array [],
        "comments": "article comments",
        "createTime": 1563697765000,
        "deleteTime": 0,
        "description": "article description",
        "enclosures": Array [],
        "feedId": "http://www.ruanyifeng.com/blog/atom.xml",
        "image": "",
        "isDayFirst": true,
        "isStarred": false,
        "isUnread": false,
        "link": "http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html",
        "originLink": "",
        "publishTime": 1563697765000,
        "summary": "article summary",
        "time": 1563697765000,
        "title": "article title",
      }
    }
    feedsMap={
      Immutable.Map {
        "http://www.ruanyifeng.com/blog/atom.xml": Object {
          "_id": "http://www.ruanyifeng.com/blog/atom.xml",
          "_rev": "5-55772bf2b4d4b74608a60ac3480334e4",
          "author": "ruanyifeng",
          "categories": Array [],
          "createTime": 1563686468437,
          "deleteTime": 0,
          "description": "Ruan YiFeng's Blog",
          "etag": "\\"11d86-58e2cc1eab031\\"",
          "favicon": "http://www.ruanyifeng.com/favicon.ico",
          "generator": "Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)",
          "language": "",
          "link": "http://www.ruanyifeng.com/blog/",
          "publishTime": 1563697765000,
          "time": 1563697765000,
          "title": "阮一峰的网络日志",
          "url": "http://www.ruanyifeng.com/blog/atom.xml",
        },
      }
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en-US",
        "formatDate": [Function],
        "formatDateToParts": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "addFeed": "Add Feed",
          "allArticlesAreReadNow": "All articles are read now.",
          "delete": "Delete",
          "doYouWantSetAllArticlesBeRead": "Do you want set all articles be read?",
          "feedUrl": "Feed Url",
          "feeds": "Feeds",
          "feedsAreUpdated": "Feeds are Updated",
          "inputKeywords": "Please input the keywords",
          "invalidFeedUrl": "Invalid Feed Url",
          "keywords": "Keywords",
          "languages": "Languages",
          "menuAllItems": "All Items",
          "menuStarred": "Starred",
          "menuSubscriptions": "Subscriptions",
          "menuUnread": "Unread",
          "noFeeds": "No Feeds",
          "noMatched": "No Matched",
          "settings": "Settings",
          "somethingWrong": "Something Wrong!",
          "title": "Breader",
          "unfoundFeed": "Unfound Feed",
          "unknown": "Unknown",
        },
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
      }
    }
    isUpdatingCurrentArticle={false}
  >
    <div
      className="article-view"
    >
      <div
        className="view-header"
      >
        <div
          className="view-header-right"
        >
          <Icon
            onClick={[Function]}
            theme="outlined"
            type="star"
          >
            <LocaleReceiver
              componentName="Icon"
            >
              <i
                aria-label="icon: star"
                className="anticon anticon-star"
                onClick={[Function]}
                tabIndex={-1}
              >
                <IconReact
                  className=""
                  type="star-o"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="star"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    key="svg-star"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                      key="svg-star-svg-0"
                    />
                  </svg>
                </IconReact>
              </i>
            </LocaleReceiver>
          </Icon>
        </div>
      </div>
      <Scrollbars
        autoHeight={false}
        autoHeightMax={200}
        autoHeightMin={0}
        autoHide={false}
        autoHideDuration={200}
        autoHideTimeout={1000}
        hideTracksWhenNotNeeded={false}
        renderThumbHorizontal={[Function]}
        renderThumbVertical={[Function]}
        renderTrackHorizontal={[Function]}
        renderTrackVertical={[Function]}
        renderView={[Function]}
        tagName="div"
        thumbMinSize={30}
        universal={false}
      >
        <div
          style={
            Object {
              "height": "100%",
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            key="view"
            style={
              Object {
                "WebkitOverflowScrolling": "touch",
                "bottom": 0,
                "left": 0,
                "marginBottom": 0,
                "marginRight": 0,
                "overflow": "scroll",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              className="view-content"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="article-info"
                data-link="http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html"
              >
                <div
                  className="article-date"
                >
                  <p>
                    Sun Jul 21 2019 16:29
                  </p>
                </div>
                <div
                  className="article-title"
                >
                  <h1>
                    article title
                  </h1>
                </div>
                <div
                  className="article-author"
                >
                  <p>
                    ruanyifeng
                     @
                     
                    阮一峰的网络日志
                  </p>
                </div>
              </div>
              <div
                className="article-content real-content"
              />
            </div>
          </div>
          <div
            key="trackHorizontal"
            style={
              Object {
                "borderRadius": 3,
                "bottom": 2,
                "display": "none",
                "height": 6,
                "left": 2,
                "position": "absolute",
                "right": 2,
              }
            }
          >
            <div
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,.2)",
                  "borderRadius": "inherit",
                  "cursor": "pointer",
                  "display": "block",
                  "height": "100%",
                  "position": "relative",
                }
              }
            />
          </div>
          <div
            key="trackVertical"
            style={
              Object {
                "borderRadius": 3,
                "bottom": 2,
                "display": "none",
                "position": "absolute",
                "right": 2,
                "top": 2,
                "width": 6,
              }
            }
          >
            <div
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,.2)",
                  "borderRadius": "inherit",
                  "cursor": "pointer",
                  "display": "block",
                  "position": "relative",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </Scrollbars>
      <div
        className="view-footer"
      >
        <p />
      </div>
      <WebviewDrawer
        onClose={[Function]}
        src=""
        visible={false}
        width="calc(100vw - 490px)"
      >
        <withConfigConsumer(Drawer)
          className="webview-drawer"
          closable={false}
          onClose={[Function]}
          visible={false}
          width="calc(100vw - 490px)"
        >
          <Drawer
            className="webview-drawer"
            closable={false}
            getPrefixCls={[Function]}
            height={256}
            keyboard={true}
            level={null}
            mask={true}
            maskClosable={true}
            onClose={[Function]}
            placement="right"
            prefixCls="ant-drawer"
            renderEmpty={[Function]}
            visible={false}
            width="calc(100vw - 490px)"
          >
            <DrawerWrapper
              afterVisibleChange={[Function]}
              className="webview-drawer"
              defaultOpen={false}
              duration=".3s"
              ease="cubic-bezier(0.78, 0.14, 0.15, 0.86)"
              forceRender={false}
              getContainer="body"
              handler={false}
              keyboard={true}
              level={null}
              maskClosable={true}
              maskStyle={Object {}}
              onChange={[Function]}
              onClose={[Function]}
              open={false}
              placement="right"
              prefixCls="ant-drawer"
              showMask={true}
              style={
                Object {
                  "transform": undefined,
                  "zIndex": undefined,
                }
              }
              width="calc(100vw - 490px)"
              wrapperClassName=""
            >
              <PortalWrapper
                forceRender={false}
                getContainer="body"
                visible={false}
                wrapperClassName=""
              />
            </DrawerWrapper>
          </Drawer>
        </withConfigConsumer(Drawer)>
      </WebviewDrawer>
      <ArticleViewSkeleton
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="article-view-skeleton "
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="view-content"
          >
            <div
              className="article-info"
            >
              <div
                className="article-date"
              >
                <SpanSkeleton
                  width={138}
                >
                  <span
                    className="span-skeleton skeleton-active "
                    style={
                      Object {
                        "width": 138,
                      }
                    }
                  />
                </SpanSkeleton>
              </div>
              <div
                className="article-title"
              >
                <TextSkeleton
                  style={
                    Object {
                      "height": 40,
                      "marginBottom": 8,
                    }
                  }
                >
                  <div
                    className="text-skeleton skeleton-active "
                    style={
                      Object {
                        "height": 40,
                        "marginBottom": 8,
                        "width": undefined,
                      }
                    }
                  />
                </TextSkeleton>
              </div>
              <div
                className="article-author"
              >
                <SpanSkeleton
                  width={156}
                >
                  <span
                    className="span-skeleton skeleton-active "
                    style={
                      Object {
                        "width": 156,
                      }
                    }
                  />
                </SpanSkeleton>
              </div>
            </div>
            <div
              className="article-content"
            >
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton
                width="57%"
              >
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": "57%",
                    }
                  }
                />
              </TextSkeleton>
              <br />
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton
                width="83%"
              >
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": "83%",
                    }
                  }
                />
              </TextSkeleton>
              <br />
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton>
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": undefined,
                    }
                  }
                />
              </TextSkeleton>
              <TextSkeleton
                width="67%"
              >
                <div
                  className="text-skeleton skeleton-active "
                  style={
                    Object {
                      "width": "67%",
                    }
                  }
                />
              </TextSkeleton>
            </div>
          </div>
        </div>
      </ArticleViewSkeleton>
    </div>
  </ArticleViewComponent>
</injectIntl(ArticleViewComponent)>
`;
